src/lucky/routable.cr

```diff
@@ -145,16 +145,17 @@ module Lucky::Routable
     # no-op by default
   end
 
-  NORMALIZED_ROUTES = [] of Nil # types are not resolved in macros
+  NORMALIZED_ROUTES = {} of String => Hash(Symbol, String)
 
   # :nodoc:
   macro enforce_route_uniqueness(method, original_path)
     # Regex for capturing the param part for normalization
     #
     # So "/users/:user_id" is changed to "/users/:normalized"
     {% normalized_path = original_path.gsub(/(\:\w*)/, ":normalized") %}
+    {% normalized_key = "#{method.id} #{normalized_path.id}" %}
 
-    {% if already_used_route = NORMALIZED_ROUTES.find { |route| route[:normalized_path] == normalized_path && route[:method] == method } %}
+    {% if already_used_route = NORMALIZED_ROUTES[normalized_key] %}
```

Comment:
If `NORMALIZED_ROUTES[normalized_key]` doesn't exist, this still works? In runtime, you'd have to add the `?` to say it's nilable otherwise it'd throw an error. I guess it works different in macro land?

Reply:
Yeah. I was equally surprised by this. See https://github.com/crystal-lang/crystal/pull/10930. 🙂